### PR TITLE
refactor: migrate models to SQLAlchemy v2

### PR DESCRIPTION
**Pull request by yuxiaoy1**
Please review @frostming @greyli 
I think we need to set some fields to optional.

### CODE REVIEW

File: backend/bamboo/models.py [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
+    name: so.Mapped[str]
+    username: so.Mapped[str]
+    password_hash: so.Mapped[str]
+    bio: so.Mapped[Optional[str]]
+    introduction: so.Mapped[Optional[str]]
+    created_at: so.Mapped[datetime] = so.mapped_column(default=lambda: datetime.now(UTC))
+    updated_at: so.Mapped[datetime] = so.mapped_column(
+        default=lambda: datetime.now(UTC), onupdate=lambda: datetime.now(UTC)
+    )
+    active: so.Mapped[bool]
```
Comment (greyli) [inline backend/bamboo/models.py:22]: @frostming Do we need to set this field defaults to `False` or `True`?

File: backend/bamboo/models.py [hunk 2]
```diff
@@ -38,251 +41,253 @@ def validate_password(self, password):
...
-    content = Column(Text)
-    start = Column(DateTime)
-    end = Column(DateTime)
-    created_at = Column(DateTime, default=datetime.utcnow)
-    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
-
-    venue = relationship("Venue", back_populates="schedule_items")
-    talk = relationship("Talk", back_populates="schedule_items")
+    venue_id: so.Mapped[int] = so.mapped_column(sa.ForeignKey("venue.id"), primary_key=True)
+    talk_id: so.Mapped[int] = so.mapped_column(sa.ForeignKey("talk.id"), primary_key=True)
```
Comment (frostming) [inline backend/bamboo/models.py:269]: This can be empty, not all schedule items should be a talk

**Review by frostming — CHANGES_REQUESTED** [review]
Left some comments

File: backend/bamboo/models.py (hunk 1 above)
Comment (frostming) [reply backend/bamboo/models.py:22]: False

**Comment by yuxiaoy1** [comment]
Updated, please review @frostming @greyli

**Review by frostming — APPROVED** [review]

**Pull request merged**